docs/api/specs/browserResidentDesktopAgents.md

```diff
@@ -0,0 +1,234 @@
+
+When an app runs `getAgent()`, it checks for the existence of `window.parent`, `window.opener` and `window.parent.opener` (collectively called "Parents"). This function sends a "Handshake" message to each Parent via a `postMessage` in the hopes of discovering a DA.
+
+If we run this logic in reverse we see that apps may be launched:
+
+(1) By creating an iframe in a DA Window
+(2) By calling `window.open` from a DA Window
+(3) By creating an iframe in a window that was opened from a DA Window
+
+> Note - This covers most typical cases but it leaves out more complex launching scenarios such as "daisy chains" (frame -> frame or open -> open) or allowing end users to open tabs in an adhoc manner. To cover these cases, DAs may choose to allow themselves to be opened as hidden iframes within app windows. See the section on hidden iframes at the end of this document.
+
+## Responding to app instance connections - Web Connection Protocol (WCP)
+
+DAs MUST run `addEventListener("message",...)` to receive incoming connection requests from apps. This will receive the initial `postMessage` "Handshake" messages from apps. Handshakes do not contain any identifying data. They contain only a "nonce" (a unique id).
+
+Upon receiving an incoming Handshake the DA MUST:
+
+1) Create a [MessageChannel](https://developer.mozilla.org/en-US/docs/Web/API/Channel_Messaging_API)
+2) Set up a processing thread to receive and process BCP requests for this connection
+3) Respond to the event's `source` window by echoing the nonce and transferring `port2` from the channel
+
+Example WCP negotiation
+```JavaScript
+window.addEventListener("message", (event) => {
+    const { origin, data, source } = event
+    const channel = new MessageChannel()
+    startBCPProcessing(event, channel) // See next step
+    source.postMessage({ nonce: data.nonce }, origin, [channel.port2])
+});
+```
+
+Alternatively for step 3, a DA MAY return a `url` instead of transferring a MessagePort. When `url` is provided in the response, `getAgent()` will open that `url` in a hidden iframe and initiate another WCP handshake with that frame.
+
+Example WCP negotiation with URL
+```JavaScript
+window.addEventListener("message", (event) => {
+    const { origin, data, source } = event
+    source.postMessage({ nonce: data.nonce, url }, origin)
+});
+```
+
+## Responding to app communications with Desktop Agent Communication Protocol (DACP)
+
+BCP processing should begin by setting up an inactive connection instance. This instance will become active after the first BCP "WCPValidateAppIdentity" message is received and processed (or deleted if it fails). It is important to remember the WindowProxy (event.source).
+
+An `instanceUuid` should be established at this point. This will be used in following steps.
+
+```JavaScript
+const startBCPProcessing(event, channel) => {
```

Comment:
this reads like we are suggesting that people implement their own da-proxy. However, I'm pretty sure that's not the case as it's solely for use on the web, right?

Reply:
All needs a rewrite which I'm on this week